### DISCUSSION TITLE
no-useless-escape issue in regexp strings

### DISCUSSION THREAD

**Discussion by erosman** [eslint/eslint#17744] [Q&A] [answered]
The backslashes in the following example show as error in ESLint with "no-useless-escape".
Can the following error be avoided?

```js
const code = 
`(() => {
  const regex = /\/\d+/g;
  if (regex.test(window.location.href)) {
    // do something
  }
})();`;

browser.tabs.executeScript({code});
```

**Comment by fasttime** [comment, marked as answer]
Thanks for the clarifications. Backslashes in template literals are considered escape characters, so the rule is correct in reporting them. They should be preceded by another backslash in order to result as a single backslash in the value that is assigned to `code`:

```js
const code = 
`(() => {
  const regex = /\\/\\d+/g;
  if (regex.test(window.location.href)) {
    // do something
  }
})();`;
```

Another option is using `String.raw` to get the unescaped text inside the backticks:

```js
const code = 
String.raw`(() => {
  const regex = /\/\d+/g;
  if (regex.test(window.location.href)) {
    // do something
  }
})();`;
```

**Reply by erosman** [reply]
Thank you for the help and the explanation.
